pkg/api/http_server.go

```diff
@@ -819,13 +820,71 @@ func (hs *HTTPServer) readCertificates() (*tls.Certificate, error) {
 		return nil, fmt.Errorf(`cannot find SSL key_file at %q`, hs.Cfg.KeyFile)
 	}
 
+	if hs.Cfg.CertPassword != "" {
+		return handleEncryptedCertificates(hs.Cfg)
+	}
+	// previous implementation
 	tlsCert, err := tls.LoadX509KeyPair(hs.Cfg.CertFile, hs.Cfg.KeyFile)
 	if err != nil {
 		return nil, fmt.Errorf("could not load SSL certificate: %w", err)
 	}
 	return &tlsCert, nil
 }
 
+func handleEncryptedCertificates(cfg *setting.Cfg) (*tls.Certificate, error) {
+	certKeyFilePassword := cfg.CertPassword
+	certData, err := os.ReadFile(cfg.CertFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read certificate file: %w", err)
+	}
+
+	keyData, err := os.ReadFile(cfg.KeyFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read private key file: %w", err)
+	}
+
+	// handle encrypted private key
+	keyPemBlock, _ := pem.Decode(keyData)
+
+	var keyBytes []byte
+	// Process the X.509-encrypted or PKCS-encrypted PEM block.
+	// nolint:staticcheck
+	if x509.IsEncryptedPEMBlock(keyPemBlock) {
```

Comment:
I am not certain we should support this method of encryption, seeing as it has been shown to be insecure, and these functions are deprecated. Is it a requirement?

Reply:
Really appreciate the feedback. I was doubting to leave this to be honest, but agree that it is not needed. If the customer actually needs it (the requirement was not that detailed unfortanelly), we will address this in a different PR. Thanks, I will remove this part